### DISCUSSION TITLE
Handling Packages that auto update

### DISCUSSION THREAD

**Discussion by Azathothas** [pkgforge/soarpkgs#173] [Ideas]
Relevant homebrew discussion: https://github.com/orgs/Homebrew/discussions/3359

Basically, we need a way to handle packages that auto update themselves, ideally these packages also change their filenames, sometimes even locations..
Homebrew has an option to force update these kind of packages; they rely on their upstream metadata & use that to upgrade/downgrade (They disregard the update done by the package itself)
Do we handle it this way?
Or is there a better way to do it

[a comment by an automated account is not shown]

**Comment by Kreijstal** [comment]
3rd option: patch them so they dont autoupgrade, use soarpkg to upgrade

**Reply by Azathothas** [reply]
Unlike homebrew, we don't pin versions, so our patch won't work if upstream changes drastically.
If we add some kind of smart/automated patching, that too will be complicated because we can't know in advance where (the update script/code) and how (replacing itself or downloading to temp dir/current dir) the affected app is updating
